worlds/hk_rework/state_mixin.py

```diff
@@ -668,35 +679,58 @@ def can_exclude(self, options):
 
 
 class EquipCharmVariable(RCStateVariable):
-    # TODO - skipped
-    prefix = "$EQUIPPEDCHARM"
-    equip_prefix = "CHARM"
-    no_equip_prefix = "noCHARM"
+    prefix: str = "$EQUIPPEDCHARM"
+    equip_prefix: str = "CHARM"
+    no_equip_prefix: str = "noCHARM"
     excluded_charm_ids: Tuple[int] = (23, 24, 25, 36,)  # fragiles and Kingsoul
     charm_id: int
     charm_name: str
+    notch_cost: int
+    charm_key: str
 
-    class EquipResult(enum.Enum):
+    class EquipResult(IntEnum):
         NONE = 1
         OVERCHARM = 2
         NONOVERCHARM = 3
 
+    def __init__(self, term: str, world: "HKWorld"):
+        super().__init__(term)
+        self.charm_id, self.charm_name = self.charm_id_and_name(term)
+        self.notch_cost = world.charm_names_and_costs[self.charm_name]
+        self.charm_key = f"CHARM{self.charm_id}"
+
+    @staticmethod
+    def get_name(charm: int | str) -> str:
+        """Convert charm id to name, or just return the name"""
+        if charm.isdigit():
+            return charm_names[charm - 1]
+        else:
+            return charm
+
+    @staticmethod
+    def get_id(charm: str | int) -> int:
+        """Convert charm name to id, or just return the id"""
+        if charm.isdigit():
+            return charm
+        else:
+            return charm_name_to_id[charm] + 1
+
     @staticmethod
-    def charm_id_and_name(charm) -> Tuple[int, str]:
+    def charm_id_and_name(charm: str | int) -> Tuple[int, str]:
```

Comment:
```suggestion
    def charm_id_and_name(charm: str) -> Tuple[int, str]:
```
the arg is always a str, it just might be "1" or "Compass"